File: plugins/methods/src/com/intellij/methods/fs/MethodsVirtualFile.java
package com.intellij.methods.fs;

import com.intellij.testFramework.LightVirtualFile;
import org.jetbrains.annotations.NotNull;

import java.io.File;

public final class MethodsVirtualFile extends LightVirtualFile {
  private final String myPath;

  public MethodsVirtualFile(@NotNull String path, @NotNull String content) {
    super(new File(path).getName(), content);
    myPath = path;
    setWritable(false);
  }

  @Override
  public @NotNull String getPath() {
    return myPath;
  }

  @Override
  public @NotNull MethodsVirtualFileSystem getFileSystem() {
    return MethodsVirtualFileSystem.getInstance();
  }
}

File: plugins/methods/src/com/intellij/methods/fs/MethodsVirtualFile.kt
package com.intellij.methods.fs

import com.intellij.testFramework.LightVirtualFile
import java.io.File

/**
 * Lightweight file backed by [MethodsVirtualFileSystem].
 */
class MethodsVirtualFile(private val myPath: String, content: String) : LightVirtualFile(File(myPath).name, content) {

    init {
        isWritable = false
    }

    override fun getPath(): String = myPath

    override fun getFileSystem(): MethodsVirtualFileSystem = MethodsVirtualFileSystem.getInstance()
}

